### PR TITLE
system-variables: add skip-name-resolve

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

Adds documentation for the PR https://github.com/pingcap/tidb/pull/25464

This is a minor feature.

### Which TiDB version(s) do your changes apply to? (Required)

master, 5.2.

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/25464

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by dveeden — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -212,6 +212,23 @@ mysql> SELECT * FROM t1;
 - Range: `[0, 65535]`
 - The port that the `tidb-server` is listening on when speaking the MySQL protocol.
 
+### skip_name_resolve <span class="version-mark">New in v5.2.0</span>
+
+- Scope: GLOBAL
+- Default value: `OFF`
+- This controls if the `tidb-server` should resolve hostnames as part of the connection handshake.
```
Comment (TomShawn) [inline system-variables.md:219]:
```suggestion
- This variable controls whether the `tidb-server` instance resolves hostnames as a part of the connection handshake.
```

File: system-variables.md [hunk 2]
```diff
@@ -212,6 +212,23 @@ mysql> SELECT * FROM t1;
...
+- Enabling this option is useful as a performance optimization when DNS is unreliable.
```
Comment (TomShawn) [inline system-variables.md:220]:
```suggestion
- When the DNS is unreliable, you can enable this option to improve network performance.
```

File: system-variables.md [hunk 3]
```diff
@@ -212,6 +212,23 @@ mysql> SELECT * FROM t1;
...
+
+> **Note:**
+>
+> When `skip_name_resolve=ON`, users with a hostname in their identity will no longer be able to log in. i.e.
```
Comment (TomShawn) [inline system-variables.md:224]:
```suggestion
> When `skip_name_resolve=ON`, users with a hostname in their identity will no longer be able to log into the server. For example:
```

File: system-variables.md [hunk 4]
```diff
@@ -212,6 +212,23 @@ mysql> SELECT * FROM t1;
...
+>
+> ```sql
+> CREATE USER 'appuser'@'apphost' IDENTIFIED BY 'app-password';
+> ```
+>
+> In this example, it is recommended to replace _apphost_ with an IP address or wildcard (`%`).
```
Comment (TomShawn) [inline system-variables.md:230]:
```suggestion
> In this example, it is recommended to replace `apphost` with an IP address or the wildcard (`%`).
```

**Review by TomShawn — COMMENTED** [review]
Some minor comment

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**